From 0af411e4d13594305d53cdfdb9a6cdd66b09ee16 Mon Sep 17 00:00:00 2001
From: Catalin Popescu <catalin.popescu@intel.com>
Date: Fri, 13 May 2016 15:29:18 +0200
Subject: [PATCH 2/2] Dollar Cove TI : handle IRQ as triggered on high level.

PMIC IRQ is triggered on rising edge. As a result, when multiple source
interrupts fire, some interrupts are missed (ex : Power Button and ADC
interrupts). This issue has been reported by customer. When this happens
PMIC IRQ stops firing and power button stops working.

Change-Id: I4516f08ac96458b3296833660080a8a32b5c016a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-32240
Signed-off-by: Catalin Popescu <catalin.popescu@intel.com>
Reviewed-on: https://android.intel.com:443/505999
---
 drivers/mfd/intel_soc_pmic_dc_ti.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mfd/intel_soc_pmic_dc_ti.c b/drivers/mfd/intel_soc_pmic_dc_ti.c
index 7ccaca6..87c44c1 100644
--- a/drivers/mfd/intel_soc_pmic_dc_ti.c
+++ b/drivers/mfd/intel_soc_pmic_dc_ti.c
@@ -166,7 +166,7 @@ static int dollar_cove_ti_init(void)
 
 struct intel_soc_pmic dollar_cove_ti_pmic = {
 	.label		= "dollar cove_ti",
-	.irq_flags	= IRQF_TRIGGER_RISING | IRQF_ONESHOT,
+	.irq_flags	= IRQF_TRIGGER_HIGH | IRQF_ONESHOT,
 	.init		= dollar_cove_ti_init,
 	.cell_dev	= dollar_cove_ti_dev,
 	.irq_regmap	= dollar_cove_ti_irqregmap,
-- 
1.9.1

